### PR TITLE
Set up sccache inside the DockerFile.

### PR DESCRIPTION
**Pull request by Jorghi12**
1) Start setting up sccache inside of the DockerFile, instead of in the jenkins build script. Also created a fork  of sccache that supports HCC, uploaded to AWS. 

2) Allowing maximum swap space usage for ROCm builds to alleviate the LLVM out of memory issue we've been facing.

### CODE REVIEW

**Review by yf225 — CHANGES_REQUESTED** [review]
See comment

**Comment by Jorghi12** [comment]
@yf225  I'll add the logic for that right now.

**Review by yf225 — APPROVED** [review]

**Comment by yf225** [comment]
I will start a Caffe2 docker image build based on the changes here.

**Comment by Jorghi12** [comment]
Spoke with @yf225. Easiest way to deploy this is to create two separate Pull Requests. First creating one for the new DockerFile and having it merged to master. Then, create one with the build script adjustment, which will make it easier to test on the new Docker Images.

**Comment by Jorghi12** [comment]
Since this change is updating the DockerFile & the build script at once, the caffe2-docker trigger https://ci.pytorch.org/jenkins/job/caffe2-docker-trigger/148/ will be the source of truth.

**Comment by Jorghi12** [comment]
The Caffe2 Docker Trigger passes all but one: https://ci.pytorch.org/jenkins/job/caffe2-docker-trigger/149/

Due to:
`denied: The repository with name 'caffe2/py2-mkl-ubuntu16.04' in registry with id '308535385114' already has the maximum allowed number of images which is '10000'`

File: docker/caffe2/jenkins/build.sh [hunk 1]
```diff
@@ -76,7 +76,7 @@ if [ -n "${JENKINS:-}" ]; then
 fi
 
 # Build image
-docker build \
+docker build $(if [[ "$image" == *rocm* ]]; then echo "--memory-swap -1"; fi) \
```
Comment (ezyang) [inline docker/caffe2/jenkins/build.sh]: Doesn't this control the swap memory available when building the docker image, not when running it?

File: docker/caffe2/jenkins/common/install_rocm.sh [hunk 2]
```diff
@@ -58,6 +63,7 @@ install_hip_thrust() {
     cd /data/Thrust/thrust/system/cuda/detail/cub-hip && git checkout hip_port_1.7.4_caffe2 && cd -
 }
 
+# This will be removed after merging an upcoming PR.
 install_hcrng() {
```
[the remark this answers is not in the document]
Comment (Jorghi12) [reply docker/caffe2/jenkins/common/install_rocm.sh]: The PyTorch rocRAND PR hasn't been merged yet. And hcrng isn't part of the ROCm distribution it seems.

**Comment by ezyang** [comment]
Can we merge this?

**Comment by ezyang** [comment]
This must be landed with an update to the Docker images.

**Comment by yf225** [comment]
@Jorghi12 Is there a new status on this one?

**Comment by ezyang** [comment]
This diff is abandoned.

**Pull request closed without merging**